src/components/DisplayProfile.jsx

```diff
@@ -1,23 +1,27 @@
 import { useEnsAddress, useNativeTransactions } from "react-moralis";
-import  getProfiles  from '@/lens/get-profiles.js'
+import getProfiles from "@/lens/get-profiles.js";
+import doesFollowFunc from '@/lens/does-follow'
 import { useState } from "react";
-
+import FollowBtn from "./FollowBtn";
+import UnfollowBtn from "./UnfollowBtn";
 
 export default function DisplayProfile(props) {
-  const [ profile, setProfile] = useState()
+  const [profile, setProfile] = useState();
+  const [doesFollow, setDoesFollow] = useState();
 
   const { name } = useEnsAddress(props.address);
   const { data: Transactions, error } = useNativeTransactions({
     address: props?.address,
   });
 
   const getProfile = async () => {
-    let { profiles } = await getProfiles({ownedBy: props?.address});
-    setProfile(profiles.items[0])
-  }
-
-  getProfile()
+    let { profiles } = await getProfiles({ ownedBy: props?.address });
+    setProfile(profiles.items[0]);
+    setDoesFollow(await doesFollowFunc(profile?.id))
+    console.log(doesFollow)
+  };
 
+  getProfile();
```

Comment:
``This one is on me, sorry. I should have noticed it on the previous review, but basically you should put your get profile function and call in a useEffect.

useEffect should be used to handle side effect, and fetching data from an API is one. 
https://reactjs.org/docs/hooks-effect.html

Also, you have the same problem as above, but this time with props.address. The first few time the component mount, the address is undefined and trigger the same error as above. You should put a condition before calling if : 

```javascript
if (props.addresss) {
   getProfile();
}
```

Overall, here is the final code I made: 
```javascript
useEffect(() => {
    async function getProfile() {
      const { profiles } = await getProfiles({ ownedBy: [props.address] });
      setProfile(profiles.items[0]);
      const resultFollow = await doesFollowFunc(profiles.items[0].id)
      setDoesFollow(resultFollow);
    };

    if (props.address) {
      getProfile();
    }
  }, [props.address])
```